frontend/src/scenes/billing/PurchaseCreditsModal.tsx

```diff
@@ -0,0 +1,202 @@
+import { LemonButton, LemonCheckbox, LemonDivider, LemonInput, LemonModal, LemonTable } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { Form } from 'kea-forms'
+import { LemonField } from 'lib/lemon-ui/LemonField'
+
+import { BillingGauge } from './BillingGauge'
+import { billingLogic } from './billingLogic'
+import { BillingGaugeItemKind } from './types'
+
+export const PurchaseCreditsModal = (): JSX.Element | null => {
+    const { showPurchaseCreditsModal, submitCreditForm } = useActions(billingLogic)
+    const { selfServeCreditOverview, isCreditFormSubmitting, creditForm, creditDiscount } = useValues(billingLogic)
+
+    return (
+        <LemonModal
+            onClose={() => showPurchaseCreditsModal(false)}
+            width="max(44vw)"
+            title="Buy credits in advance, get a discount"
+            footer={
+                <>
+                    <LemonButton
+                        type="secondary"
+                        onClick={() => showPurchaseCreditsModal(false)}
+                        disabled={isCreditFormSubmitting}
+                    >
+                        Cancel
+                    </LemonButton>
+                    <LemonButton type="primary" onClick={() => submitCreditForm()} loading={isCreditFormSubmitting}>
+                        Buy {creditForm.creditInput ? `$${creditForm.creditInput.toLocaleString()}` : ''} credits
+                    </LemonButton>
+                </>
+            }
+        >
+            <Form formKey="creditForm" logic={billingLogic} enableFormOnSubmit>
+                <div className="flex flex-col gap-3.5">
+                    <p className="mb-0">
+                        We're giving you the option to buy credits in advance at discount of up to 30%.
+                    </p>
+
+                    <p className="mb-0">
+                        Based on your usage, we think you'll need{' '}
+                        <b>
+                            $
+                            {(+selfServeCreditOverview.estimated_monthly_credit_amount_usd * 12).toLocaleString(
+                                'en-US',
+                                {
+                                    minimumFractionDigits: 0,
+                                    maximumFractionDigits: 0,
+                                }
+                            )}
+                        </b>{' '}
+                        credits this year. That's{' '}
+                        <b>
+                            $
+                            {(+selfServeCreditOverview.estimated_monthly_credit_amount_usd).toLocaleString('en-US', {
+                                minimumFractionDigits: 0,
+                                maximumFractionDigits: 0,
+                            })}
+                        </b>{' '}
+                        per month.
+                    </p>
+
+                    <LemonField name="creditInput" label="How many credits do you want to purchase?">
```

Comment:
Actually I think we can put some subtext here also. smaller, text-muted

"Credits are dispersed monthly and roll over to the next month. If you use more than the available credits in any month, you'll pay for the usage at the standard rate. Credits expire after 1 year from purchase."

Reply:
Agree this good to add.